aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -611,19 +605,38 @@ TensorIterator TensorIterator::reduce_op(Tensor& out1, Tensor& out2, const Tenso
 void TensorIterator::mark_outputs() {
   for (int i = 0; i < num_outputs_; i++) {
     operands_[i].is_output = true;
-    const auto &output = operands_[i].tensor;
+    const auto& output = operands_[i].tensor;
     if (!output.defined()) continue;
 
     // check if output is also an input
     for (int arg = num_outputs_; arg < ntensors(); arg++) {
-      const auto &input = operands_[arg].tensor;
+      const auto& input = operands_[arg].tensor;
       if (output.is_same(input)) {
         operands_[i].is_read_write = true;
       }
     }
   }
 }
 
+void TensorIterator::check_mem_overlaps() {
+  if (check_mem_overlap_) {
+    for (int i = 0; i < ntensors(); i++) {
```

Comment:
We can make this code less repetitive by doing something like the following:
```
      for (int i = 0; i < num_outputs_; i++) {
        const auto& output = operands_[i].tensor;
        if (!output.defined()) continue;
        assert_no_internal_overlap(output);
        for (int j = num_outputs_; j < ntensors(); j++) {
          const auto& input = operands_[j].tensor;
          assert_no_partial_overlap(output, input);
        }
      }
```

Reply:
addressed